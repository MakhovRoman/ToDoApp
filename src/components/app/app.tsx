import React from 'react';
import 'styles/index.scss';
import { TodoList } from '../todo-list/TodoList';
import { TodoPanel } from '../todo-panel/TodoPanel';
import { TodoModal } from '../todo-modal/TodoModal';

export const App = () => {
  return (
    <div className='wrapper' data-testid='app'>
      <h1>TODO LIST</h1>
      <div className='content'>
        <TodoPanel />
        <TodoList />
        <TodoModal />
      </div>
    </div>
  )
}
