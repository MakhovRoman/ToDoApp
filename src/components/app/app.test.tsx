import { fireEvent, render, screen} from '@testing-library/react';
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom';
import renderer from "react-test-renderer";
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { App } from './app';
import { Store, AnyAction } from '@reduxjs/toolkit'


let domTree: renderer.ReactTestRenderer;
let store: Store<unknown, AnyAction>

beforeAll(() => {
  const mockStore = configureStore();
    const initialState = {
      todoList: {
        todoList: []
      },
      todoSelect: {
        select: "all"
      },
      setModalOpen: {
        isOpen: false
      }
    }
    store = mockStore(initialState);
});

describe("Components/App", () => {
  it("Matches DOM Snapshot", () => {
    domTree = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    );

    expect(domTree).toMatchSnapshot();
  });

  it("Open modal", async () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>
    )

    const button = screen.getByTestId("button-openModal");

    expect(screen.queryByText("Title")).toBeNull();

    await userEvent.click(button);

    expect(screen.queryByText("Title")).toBeDefined();
  })
})
