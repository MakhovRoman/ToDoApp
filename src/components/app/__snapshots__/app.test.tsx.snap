// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/App Matches DOM Snapshot 1`] = `
<div
  className="wrapper"
  data-testid="app"
>
  <h1>
    TODO LIST
  </h1>
  <div
    className="content"
  >
    <div
      className="todoPanel"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
        data-testid="button-openModal"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Add Task
      </button>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary  css-lxf8mu-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
        onClick={[Function]}
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="button"
          tabIndex={0}
        >
          All
        </div>
        <input
          aria-hidden={true}
          aria-invalid={false}
          className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          onAnimationStart={[Function]}
          onChange={[Function]}
          tabIndex={-1}
          value="all"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-ihdtdm"
          >
            <span
              className="notranslate"
            >
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      className="todoList"
    >
      <span
        className="todoList_empty"
      >
        No Todo Found
      </span>
    </div>
    <div />
  </div>
</div>
`;
